### ISSUE TITLE
Unable click and fill up security code and expiration date using Microsoft Edge

### ISSUE THREAD

**Issue by crow1796** [recurly/recurly-js#509]
Browser: Microsoft Edge
![image](https://user-images.githubusercontent.com/12120770/51949635-49828500-2468-11e9-90ed-70f06a7b227e.png)

Console error:
![image](https://user-images.githubusercontent.com/12120770/51949661-615a0900-2468-11e9-9079-48ac8a95fefd.png)

**Comment by chrissrogers** [comment]
Hi @crow1796 -- thanks very much for providing a screenshot and relevant details for the error.

Due to PCI compliance restrictions inherent to the design of Recurly.js card fields, it is not possible to fill in any hosted fields automatically. This constraint is due to the need to isolate those fields and values from the parent page, preventing manipulation or capture of payment information.

If you could describe your use case and desired outcome, I would be happy to suggest an alternative approach if any is available.

**Comment by crow1796** [comment]
@chrissrogers I'm filling it up automatically. All of our customers can't proceed with the payment because they can't fill in their payment information, but it only started after the latest Microsoft Edge Update (Microsoft Edge v44.17763.1.0).

**Comment by chrissrogers** [comment]
I see. Do you have a link to your payment form? Since the error is coming from Vue but appears to be related to postMessage communication, I'd like to observe it myself to do some debugging.

**Comment by crow1796** [comment]
@chrissrogers https://lawformsusa.com/ try to register then select any plan, and then you'll be redirected to the payment page (Don't pay, just try to click and fill in the expiration and CVV Fields), please test it ASAP because we'll be going to apply the temporary fix/solution that we made, Thanks!

**Comment by chrissrogers** [comment]
Were you able to resolve the issue? I was able to successfully input values into the form using Edge 44.17763.1.0

![](https://cldup.com/tnMmu2bKQ0-3000x3000.png)

**Comment by geoffdutton** [comment]
Wait, so what's the temp fix? We're seeing the same thing in 44.

**Comment by chrissrogers** [comment]
It appears @crow1796 modified recurly.js to add a try-catch to the `Bus.add` method, and has self-hosted the modified library. Our fix for the issue introduced in 44 is incoming -- in the meantime, his modification is a viable stopgap.

**Comment by crow1796** [comment]
@geoffdutton we've modified and the recurly.js file and wrapped the `line 4559` in a `try-catch`.

```
try {
    ~this.recipients.indexOf(e) || (this.recipients.push(e), "function" == typeof e.emit && e.emit("bus:added", this), this.debug("added recipient", e))
} catch(e) {
    this.debug("Edge Error: ", e);
}
```

Thanks @chrissrogers!

**Comment by geoffdutton** [comment]
@crow1796 awesome thanks!

I ended up borrowing your recurly.js (hosting on our server though), hope you don’t mind 😉

**Comment by chrissrogers** [comment]
Really glad you've both come up with quick solutions while we move through the process of issuing the fix. I'll alert here once the new release (including the EdgeHTML 18 bugfix) is on the CDN.

**Comment by crow1796** [comment]
> @crow1796 awesome thanks!
> 
> I ended up borrowing your recurly.js (hosting on our server though), hope you don’t mind 😉

Yeah sure @geoffdutton :)

**Comment by chrissrogers** [comment]
Hi @crow1796 @geoffdutton. We have just released [v4.9.5](https://github.com/recurly/recurly-js/releases/tag/v4.9.5) to our CDN, which includes the aforementioned fix for EdgeHTML18 init errors. Please feel free to test it out in your scenarios and let me know how things go!

**Comment by crow1796** [comment]
Cool, thanks @chrissrogers !
